packages/components/pie-switch/test/component/pie-switch.spec.ts

```diff
@@ -1,194 +1,196 @@
-                const component = await mount(PieSwitch, {
-                    props: {
-                        label: 'Label',
-                        labelPlacement: 'leading',
-                    } as SwitchProps,
-                });
+                const switchPage = new BasePage(page, 'switch');
+                const props: Partial<SwitchProps> = {
+                    label: 'Label',
+                    labelPlacement: 'leading',
+                };
+
+                await switchPage.load({ ...props });
 
                 // Act
-                await page.click(switchLabelSelector());
+                await page.getByTestId(pieSwitch.selectors.label.leading.dataTestId).click();
 
-                const pieSwitchComponent = await component.locator(componentSelector).isChecked();
+                const pieSwitchComponent = await page.getByTestId(pieSwitch.selectors.container.dataTestId).isChecked();
 
                 // Assert
                 expect(pieSwitchComponent).toBe(true);
             });
 
-            test('should set `checked` to `false`', async ({ mount, page }) => {
+            test('should set `checked` to `false`', async ({ page }) => {
                 // Arrange
-                const component = await mount(PieSwitch, {
-                    props: {
-                        label: 'Label',
-                        labelPlacement: 'leading',
-                        checked: true,
-                    } as SwitchProps,
-                });
+                const switchPage = new BasePage(page, 'switch');
+                const props: Partial<SwitchProps> = {
+                    label: 'Label',
+                    labelPlacement: 'leading',
+                    checked: true,
+                };
+
+                await switchPage.load({ ...props });
 
                 // Act
-                await page.click(switchLabelSelector());
+                await page.getByTestId(pieSwitch.selectors.label.leading.dataTestId).click();
 
-                const pieSwitchComponent = await component.locator(componentSelector).isChecked();
+                const pieSwitchComponent = await page.getByTestId(pieSwitch.selectors.container.dataTestId).isChecked();
 
                 // Assert
                 expect(pieSwitchComponent).toBe(false);
             });
 
-            test('should emit an event with the correct name', async ({ mount, page }) => {
+            test('should emit an event with the correct name', async ({ page }) => {
                 // Arrange
-                let eventHeard = false;
-                await mount(PieSwitch, {
-                    on: {
-                        [ON_SWITCH_CHANGED_EVENT]: () => {
-                            eventHeard = true;
-                        },
-                    },
+                const switchPage = new BasePage(page, 'switch');
+                await switchPage.load();
+
+                const expectedEventMessage = 'Switch clicked';
```

Comment:
I think this constant could be shared between files - it isn't clear from this test why that string should appear

Reply:
I think for this one, it's only used twice between 2 tests so its fine to have them there. I could rename the tests that use them to something like, `should emit a "Switch clicked" console event when the switch is clicked`?

Reply:
I think what Jamie is suggesting is having a constants file in say `test` file that does:

```
// test/constant.js
export const EXPECTED_EVENT_MESSAGE = Switch clicked
```

Then import this constant in both the spec file, as well as the `pie-storybook/stories/test/pie-switch.test.stories.ts`